custom_components/waste_collection_schedule/waste_collection_schedule/source/ville_quebec_qc_ca.py

```diff
@@ -0,0 +1,258 @@
+                if match:
+                    month_name = match.group(1).lower()
+                    year = int(match.group(2))
+                    month = MONTH_NAMES.get(month_name)
+
+                    if not month:
+                        continue
+
+                    # Get all table rows (skip header row)
+                    rows = table.find_all('tr')[1:]  # Skip header row
+
+                    for row in rows:
+                        cells = row.find_all('td')
+
+                        for cell in cells:
+                            # Check if cell has a date
+                            date_element = cell.find('p', {'class': 'date'})
+                            if date_element and date_element.text.strip().isdigit():
+                                day = int(date_element.text.strip())
+
+                                # Check if cell has collection icons
+                                img_container = cell.find(
+                                    'p', {'class': 'img'})
+                                if img_container:
+                                    images = img_container.find_all('img')
+
+                                    for img in images:
+                                        french_collection_type = img.get(
+                                            'title', img.get('alt', '')).strip()
+
+                                        if french_collection_type:
+                                            # Map French collection type to English
+                                            english_collection_type = TYPE_MAP.get(
+                                                french_collection_type, french_collection_type)
+
+                                            # Create a date object
+                                            try:
+                                                collection_date = datetime.date(
+                                                    year, month, day)
+
+                                                # Add to entries list with English type and appropriate icon
+                                                entries.append(
+                                                    Collection(
+                                                        date=collection_date,
+                                                        t=english_collection_type,
+                                                        icon=ICON_MAP.get(
+                                                            french_collection_type)
+                                                    )
+                                                )
+                                            except ValueError:
+                                                # Skip invalid dates
+                                                pass
+            except Exception:
+                # Skip this calendar if parsing fails
+                continue
+
+        return entries
```

Comment:
That's pretty ugly please try to reduce your indentation level a bit  mybe thoug if not: continue instead of just if or by refactoring into multiple functions (is probably anyway a good idea to split this 200 line function into smaller parts)